blog/_posts/2021-03-05-RSEshowcaseRAMPUA.md

```diff
@@ -0,0 +1,62 @@
+
+
+#### 3. Using Sphinx and GitHub pages to create an automated documentation webpage  
+  Documenting your code is another key reproducible research habit its worth fostering. With this project the aim was to include documentation to allow anyone to take this code and run it on their own system and replicate results generated through experimentation. [Sphinx](https://www.sphinx-doc.org/en/master/) is a documentation generator commonly used alongside [readthedocs](https://docs.readthedocs.io/en/stable/index.html). As researchers wrote the model they included [docstrings](https://www.python.org/dev/peps/pep-0257/) within the code which we wanted to build into a documentation page for the project. The [autodocs](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) Sphinx extension makes this easy, allowing for docstrings to be extracted from code and rendered into reStructured Text which is used by Sphinx. 
+  Next, we wanted to automate the generation of documentation, to achieve this we looked to [GitHub actions](https://github.com/features/actions) and [GitHub pages](https://pages.github.com/). GitHub pages lets us host a simple static website associated with the code repository (just what we needed) and GitHub actions allows us to create automated workflows that run on specific conditions. In our case we adapted the workflow ([view our workflow](https://github.com/Urban-Analytics/RAMP-UA/blob/master/.github/workflows/deploy-docs.yml)) for building Sphinx documentation and then utilized an existing action that deploys a specific directory to the `gh-pages` branch (which is where GitHub pages are rendered from). This allows for the documentation to build automatically every time new code is pushed to the `master` branch and be deployed to GitHub pages helping researchers just write their code and associated docstrings and worry less about writing out more documentation.
+
+#### 4. Holding weekly developer stand ups
+
+  Working with a team distributed across the UK and all working on different aspects of the code meant it was crucial to instigate good practices around distributed team working. At the core of this was organising a weekly stand-up Microsoft Teams Call where everyone working on the code base would join for just half an hour, once a week to go through what they'd been up to, what their plans were and to identify anything blocking or potentially blocking their progress. This helped everyone keep pace with the project and make sure others were able to know what was happening and whether there was anything they could contribute. To aid with this we initially used the [Teams feature](https://github.com/team) within GitHub organisations but more on that in [6.](./#6-making-the-most-out-of-github-repository-tools)
```

Comment:
The link at the end of the paragraph does not work - appears to be no bookmark.

Reply:
Good spot, have resolved this in 17181d5